static/js/Story.jsx

```diff
@@ -127,8 +127,93 @@ StorySheetList.propTypes = {
     toggleSignUpModal: PropTypes.func
 };
 
+function capitalizeWords(str) {
+  const words = str.split(' ');
+
+  const capitalizedWords = words.map(word => {
+    return word.charAt(0).toUpperCase() + word.slice(1);
+  });
+  const result = capitalizedWords.join(' ');
+
+  return result;
+}
+const reviewStateToClassNameMap = {
+    "reviewed": "reviewed",
+    "not reviewed": "notReviewed",
+    "edited": "edited"
+}
+const reviewStateToDisplayedTextMap = {
+    "reviewed": "Reviewed",
+    "not reviewed": "Not Reviewed",
+    "edited": "Edited"
+}
+const ReviewStateIndicator = ({reviewState, callBack}) => {
+    if (!reviewState){
+        return null}
+    let reviewStateClassName = reviewStateToClassNameMap[reviewState];
+    let displayedText = reviewStateToDisplayedTextMap[reviewState];
+    return (
+    <div className={`button extraSmall reviewState ${reviewStateClassName}`} onClick={callBack}>
+        {displayedText}
+    </div>)
+
+};
+
+const PromptWrapper = ({text}) => {
+    let promptComponent = null;
+    if (Sefaria.interfaceLang == "english"){
+        const isPromptPublished = text.descriptions?.en?.published
+        if (isPromptPublished !== false || Sefaria.is_moderator){
+        promptComponent = <div className={"systemText learningPrompt"}>
+            <InterfaceText text={{"en": text.descriptions?.en?.prompt, "he": text.descriptions?.he?.prompt}} />
+        </div>
+            }
+    }
+    else if (Sefaria.interfaceLang == "hebrew"){
+        const isPromptPublished = text.descriptions?.he?.published
+        if (isPromptPublished !== false || Sefaria.is_moderator){
+        promptComponent = <div className={"systemText learningPrompt"}>
+            <InterfaceText text={{"en": text.descriptions?.en?.prompt, "he": text.descriptions?.he?.prompt}} />
+        </div>
+            }
+    }
+    return promptComponent
+}
+const ReviewStateWrapper = ({topic, text}) => {
+    let reviewIndicatorComponent = null;
+    const [reviewStateEn, setReviewStateEn] = useState(text.descriptions?.en?.review_state);
+    const [reviewStateHe, setReviewStateHe] = useState(text.descriptions?.he?.review_state);
+    const markReviewed = function(){
+        console.log("markReviewed")
+        let lang = Sefaria.interfaceLang == "english" ? 'en' : 'he';
+        let setFun = Sefaria.interfaceLang == "english" ? setReviewStateEn : setReviewStateHe;
+        let postData = {"topic": topic, "is_new": false, 'new_ref': text.ref, 'interface_lang': Sefaria.interfaceLang};
+        postData.description = text.descriptions[lang]
+        postData.description["review_state"] = 'reviewed'
```

Comment:
This can also be in the initial definition using the object spread operator (`{description: {...text.descriptions[lang], review_state: 'reviewed'}`)